src/zh/komiic/src/eu/kanade/tachiyomi/extension/zh/komiic/Komiic.kt

```diff
@@ -0,0 +1,291 @@
+            query = QUERY_SEARCH,
+        ).toJsonRequestBody()
+        return POST(queryAPIUrl, headers, payload)
+    }
+
+    override fun fetchSearchManga(
+        page: Int,
+        query: String,
+        filters: FilterList,
+    ): Observable<MangasPage> {
+        return if (query.startsWith(PREFIX_ID_SEARCH)) {
+            val mangaId = query.substringAfter(PREFIX_ID_SEARCH)
+            client.newCall(comicByIDRequest(mangaId))
+                .asObservableSuccess()
+                .map(::parseComicByID)
+        } else {
+            super.fetchSearchManga(page, query, filters)
+        }
+    }
+
+    override fun searchMangaParse(response: Response) = parseSearchList<SearchResponse>(response)
+
+    // Comic details
+    override fun mangaDetailsRequest(manga: SManga) = comicByIDRequest(manga.url.substringAfterLast("/"))
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val res = response.parseAs<Data<ComicByIDResponse>>()
+        val comic = res.data.comic.toSManga()
+        return comic
+    }
+
+    override fun getMangaUrl(manga: SManga) = "$baseUrl${manga.url}"
+
+    /**
+     * 解析日期
+     * Parse date
+     */
+    private fun parseDate(dateStr: String): Long {
+        val sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US)
+        sdf.timeZone = TimeZone.getTimeZone("UTC")
+        return sdf.parse(dateStr)?.time ?: 0L
+    }
+
+    /**
+     * 解析章節列表
+     * Parse chapter list
+     */
+    private inline fun <reified T : ChaptersResult> parseChapterList(response: Response, comicUrl: String): List<SChapter> {
+        val res = response.parseAs<Data<T>>()
+        val comics = res.data.chapters
+
+        val tChapters = comics.filter { it.type == "chapter" }
+        val tBooks = comics.filter { it.type == "book" }
+
+        val entries = (tChapters + tBooks).map { chapter ->
+            SChapter.create().apply {
+                url = "$comicUrl/chapter/${chapter.id}/page/1"
+                name = when (chapter.type) {
+                    "chapter" -> "第 ${chapter.serial} 話"
+                    "book" -> "第 ${chapter.serial} 卷"
+                    else -> chapter.serial
+                }
+                date_upload = parseDate(chapter.dateCreated)
+                chapter_number = chapter.serial.toFloatOrNull() ?: -1f
+            }
+        }.reversed()
+
+        return entries
+    }
+
+    // Chapter list
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val payload = Payload(
+            operationName = "chapterByComicId",
+            variables = ChapterByComicIdVariables(manga.url.substringAfterLast("/")),
+            query = QUERY_CHAPTER,
+        ).toJsonRequestBody()
+
+        return client.newCall(POST(queryAPIUrl, headers, payload))
+            .asObservableSuccess()
+            .map { response ->
+                parseChapterList<ChaptersResponse>(response, manga.url)
+            }
+    }
```

Comment:
override chapterListRequest method instead

Reply:
and api doesn't return the comic id. so i must get this id in manga.url

Reply:
yes, you set the `chapterListRequest` method to return the POST request; then set the `chapterListParse` to parse the response

Reply:
chapterListParse's response.request.url only have query url. does't have manga.url

Reply:
oh right didn't see that... you can put it as after a # in apiurl and access it as fragment